test/e2e/azure_test.go

```diff
@@ -40,13 +40,14 @@ import (
 
 var _ = Describe("Workload cluster creation", func() {
 	var (
-		ctx           = context.TODO()
-		specName      = "create-workload-cluster"
-		namespace     *corev1.Namespace
-		cancelWatches context.CancelFunc
-		result        *clusterctl.ApplyClusterTemplateAndWaitResult
-		clusterName   string
-		specTimes     = map[string]time.Time{}
+		ctx               = context.TODO()
+		specName          = "create-workload-cluster"
+		namespace         *corev1.Namespace
+		cancelWatches     context.CancelFunc
+		result            *clusterctl.ApplyClusterTemplateAndWaitResult
+		clusterName       string
+		additionalCleanup func()
```

Comment:
Added to enable a spec to add a func to clean up resources it makes outside of CAPZ.